test/Semantics/select-rank.f90

```diff
@@ -0,0 +1,216 @@
+!Tests for SELECT RANK Construct(R1148)
+
+   !No error expected
+   subroutine CALL_ME(x)
+    implicit none
+    integer :: x(..)
+    SELECT RANK(x)
+    RANK (0)
+      print *, "PRINT RANK 0"
+    RANK (1)
+      print *, "PRINT RANK 1"
+    END SELECT
+   end
+
+   subroutine CALL_ME9(x)
+    implicit none
+    integer :: x(..)
+    boo: SELECT RANK(x)
+    RANK (1+0)
+      print *, "PRINT RANK 1"
+    END SELECT boo
+   end subroutine
+
+   !Error expected
+   subroutine CALL_ME2(x)
+    implicit none
+    integer :: x(..)
+    integer :: y(3),j
+    !ERROR: Selector 'y' is not an assumed-rank array variable
+    SELECT RANK(y)
+    RANK (0)
+      print *, "PRINT RANK 0"
+    RANK (1)
+      print *, "PRINT RANK 1"
+     END SELECT
+
+    SELECT RANK(x)
+    RANK(0)
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 0)) ! will fail when RANK(x) is not zero here
```

Comment:
You should use this compilation-time testing technique (or another) in place of (or in addition to) the `PRINT` statements elsewhere so that the tests confirm that the right rank is visible in each case.  You should also test that the other attributes of the `SELECT RANK` construct entities are preserved (e.g., `ALLOCATABLE` entities remain allocatable, `POINTER` assumed-rank objects can be associated with a pointer assignment, and so on.)

Reply:
Test cases that depend on the shape of the array should also be present.

Reply:
Added shape related tests, also used compile-time checking test wherever necessary. I have skipped some redundant locations where we could have added the compile-time check.